src/main/scala/org/apache/spark/metrics/source/custom/AccumulatorMetricsSource.scala

```diff
@@ -0,0 +1,54 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+package org.apache.spark.metrics.source.custom
```

Comment:
I see we're overriding a class that's private to spark. How worried should we be about that?

We should provide some clear commentary here about why we're putting a class under the spark package name and the potential danger of doing so.

Reply:
:+1: adding a comment.
It's not dangerous, we're extending `Source` trait here - if its API changes in the future, compiler will warn us.